client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx

```diff
@@ -94,7 +95,9 @@ const DomainsStep: Step = function DomainsStep( { navigation, flow } ) {
 			} );
 			dispatch( submitDomainStepSelection( suggestion, getAnalyticsSection() ) );
 
-			setHideFreePlan( Boolean( suggestion.product_slug ) || shouldHideFreePlan );
+			if ( ! isOnboardingPMFlow( flow ) ) {
+				setHideFreePlan( Boolean( suggestion.product_slug ) || shouldHideFreePlan );
+			}
```

Comment:
I have two concerns here. 🙂 

1. I'm wondering if we should implement flow-specific logic in the step. As an alternative, we can maybe add `setHideFreePlan( false )` here? https://github.com/Automattic/wp-calypso/blob/3d6ad33abc422a31f3e1403a19f160064292b6aa/client/landing/stepper/declarative-flow/onboarding-pm.ts#L79-L81
2. I'm not 100% sure, but I was wondering if this statement `setHideFreePlan( Boolean( suggestion.product_slug ) || shouldHideFreePlan );` is correct.
From what I understand, if the user selects a paid domain, the domains step will hide the free plan. I'm not sure if this is the default behaviour we want for the domains step. This would probably need some investigation, but it seems like we should be able to change it to just `setHideFreePlan( shouldHideFreePlan )`.

Reply:
Great point! I've removed the code from the step in https://github.com/Automattic/wp-calypso/pull/79109/commits/bc3cce4dd97fc4fb666d8bc73af2d0debe5c1989 and moved it into the flow as suggested in https://github.com/Automattic/wp-calypso/pull/79109/commits/533ed8d61067704b40faa58519a26d6087970096.

Reply:
> I'm not 100% sure, but I was wondering if this statement setHideFreePlan( Boolean( suggestion.product_slug ) || shouldHideFreePlan ); is correct.

You are right; I had similar confusion when I read this. However, I decided to leave it intact since things will likely change given the refactor happening right now and whether we want to roll out the paid domain + Free plan modal to all the flows. We can surely revisit this when these two things settled :)